.github/workflows/run-performance.yaml

```diff
@@ -23,26 +23,113 @@ on:
+          # sudo add-apt-repository -y ppa:rmescandon/yq
+          # sudo apt-get update
+          # sudo apt-get install -y yq
+          curl -L https://github.com/mikefarah/yq/releases/download/v4.40.7/yq_linux_amd64 -o yq
+          chmod +x yq
+          sudo mv ./yq /usr/local/bin/yq
+
       - name: Template network
         run: |
           set -exo pipefail
           export RUST_CERAMIC_IMAGE=${{ inputs.rust_ceramic_image }}
           export SCENARIO_IMAGE=${{ inputs.scenario_image }}
           export JS_CERAMIC_IMAGE=${{ inputs.js_ceramic_image }}
           export TEST_NAME=${{ inputs.test_name }}
-          ls -l tests/performance/$TEST_NAME.yaml
+          ls -l performance/$TEST_NAME
+          export THIS_TEST=${{ inputs.test_name }}-${{ github.run_id }}
+          mkdir -p performance/$THIS_TEST
+          # set the network name yaml key to the test names
+          yq e '.metadata.name = env(THIS_TEST)' performance/$TEST_NAME/network.yaml \
+            > performance/$THIS_TEST/network.yaml
+          yq e '.spec.ceramic[0].image = env(JS_CERAMIC_IMAGE)' -i performance/$THIS_TEST/network.yaml
+          yq e '.spec.ceramic[0].ipfs.rust.image = env(RUST_CERAMIC_IMAGE)' -i performance/$THIS_TEST/network.yaml
+          cat performance/$THIS_TEST/network.yaml
+          echo "THIS_TEST=$THIS_TEST" >> $GITHUB_ENV
+          echo "THIS_TEST_NAMESPACE=keramik-${THIS_TEST}" >> $GITHUB_ENV
+          echo "TEST_NAME=${TEST_NAME}" >> $GITHUB_ENV
+
+      - name: Setup GKE auth
+        uses: 'google-github-actions/auth@v1'
+        with:
+          credentials_json: ${{ secrets.GKE_SA_KEY }}
+
+      - name: Get GKE credentials
+        uses: 'google-github-actions/get-gke-credentials@v1'
+        with:
+          cluster_name: ${{ vars.GKE_CLUSTER }}
+          location: ${{ vars.GKE_ZONE }}
+
+      - name: Deploy network
+        run: |
+          kubectl apply -f performance/$THIS_TEST/network.yaml
+
+      - name: Wait for bootstrap to complete
+        timeout-minutes: 8
+        run: |
+          set -exo pipefail
+          sleep 60
+          kubectl wait --for=condition=ready \
+            --timeout=240s \
+            pod \
+            -l app=ceramic \
+            -n ${THIS_TEST_NAMESPACE}
+          sleep 60
+          kubectl wait --for=condition=complete \
+            --timeout=120s \
+            job/bootstrap \
+            -n ${THIS_TEST_NAMESPACE}
+
+      - name: Template simulation
+        run: |
+          set -exo pipefail
+          yq e '.metadata.namespace = env(THIS_TEST_NAMESPACE)' performance/$TEST_NAME/simulation.yaml \
+            > performance/$THIS_TEST/simulation.yaml
+          cat performance/$THIS_TEST/simulation.yaml
+
+      - name: Run simulation
+        run: |
+          set -exo pipefail
+          kubectl apply -f performance/$THIS_TEST/simulation.yaml
+          SIMULATION_RUNTIME=$(yq e '.spec.runTime' performance/$THIS_TEST/simulation.yaml)
```

Comment:
Hmm... These steps could be run using the test driver as well, right? I can help work this out, let me know. Would be nice to be able to reuse it since it does much of this already.

Reply:
Yeah, these def could be done more reliably in the operator. The timing is really annoying and the most likely part to fail

Reply:
Ah I see, want to chat about this quickly on Monday? I don't mind doing it via scripts either.